src/components/pages/content/ContentObject.js

```diff
@@ -583,79 +585,97 @@ class ContentObject extends React.Component {
+            name: this.state.newNonOwnerCapName
           });
         }}
       >
         <div className="form-content">
-          <label htmlFor="publicKey">Public Key</label>
-          <textarea
-            name="publicKey"
+          <label htmlFor="userName">User Name</label>
+          <input
+            name="userName"
             required
-            placeholder="Public Key..."
-            onChange={event => this.setState({newOwnerCapPublicKey: event.target.value})}
+            placeholder="User name..."
+            onChange={event => this.setState({newNonOwnerCapName: event.target.value})}
           />
           <label htmlFor="publicAddress">Public Address</label>
           <input
             name="publicAddress"
             required
             placeholder="Public Address..."
-            onChange={event => this.setState({newOwnerCapPublicAddress: event.target.value})}
+            onChange={event => this.setState({newNonOwnerCapPublicAddress: event.target.value})}
+          />
+          <label htmlFor="publicKey" className="align-top">Public Key</label>
+          <textarea
+            name="publicKey"
+            required
+            placeholder="Public Key..."
+            onChange={event => this.setState({newNonOwnerCapPublicKey: event.target.value})}
           />
         </div>
       </Form>
     </Modal> : null;
 
-    const ownerCapKey = `eluv.caps.iusr${AddressToHash(this.props.objectStore.currentAccountAddress)}`;
+    let deleteCapButton;
+    let addCapsButton;
+    if(this.props.objectStore.object.isOwner) {
+      deleteCapButton = (
+        <LabelledField>
+          <Action
+            type="button"
+            className="danger action-compact action-wide"
+            onClick={() => this.DeleteCap(capAddress)}
+          >
+            Delete encryption key
+          </Action>
+        </LabelledField>
+      );
+
+      addCapsButton = (
+        <LabelledField>
+          <Action
+            type="button"
+            className="secondary"
+            onClick={() => this.setState({showNonOwnerCapManagement: true})}
+          >
+            Add encryption key
+          </Action>
+        </LabelledField>
+      );
+    }
 
     return (
-      <div className="owner-caps-container">
-        <ToggleSection label="Advanced" toggleOpen={this.state.showOwnerCapManagement}>
-          {Object.keys(this.props.objectStore.object.meta).filter(key => (key !== ownerCapKey && key.startsWith("eluv.caps.iusr"))).map(userCap => (
-            <div key={`${userCap}-caps-version-${this.state.capsVersion}`} className="owner-keys-container">
-              <LabelledField label="Key">
-                {userCap}
+      <div className="non-owner-caps-container">
+        <ToggleSection label="Encryption Keys" toggleOpen={this.state.showNonOwnerCapManagement}>
+          {Object.keys(this.props.objectStore.object.meta.owner_caps).map(capAddress => (
```

Comment:
this fails if owner_caps is undefined. Always have a fallback initialization on things like .map or Object.keys()